### PR TITLE
feat(table-row): Add alignment property

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #8958 

## Summary
Adds an `alignment` property to the Table Row component. When set - aligns slotted Table Header and Table Cell components to this value. Number and Selection cells (internally rendered) will always be vertically centered / middle aligned.

Adjusts padding of Table Header when not slotted in the `table-header` slot to better align with sibling slotted Table Cell components.

### CODE REVIEW

**Comment by macandcheese** [comment]
cc @SkyeSeitz @ashetland this does result in slight visual diff - when the Header are in a row with table cell, we no longer apply the 1.5x padding the "table header th" have. FYI and to confirm.

File: packages/calcite-components/src/components/table-cell/table-cell.scss [hunk 1]
```diff
@@ -11,11 +11,11 @@
   @apply contents;
 }
 
-:host([alignment="center"]) td {
+:host([alignment="center"]) td:not(.selection-cell):not(.number-cell) {
```
Comment (driskull) [inline packages/calcite-components/src/components/table-cell/table-cell.scss:14]: nitpick: I don't love these selectors.

Every time we add another td this selector needs to grow. I know we have a ton of these in the codebase but IMO we should discuss moving this logic into JS and use a css class to represent what this should be styling.

**Review by driskull — APPROVED** [review]
This looks good to me. I just think sometimes our CSS selectors can get quite complex. 😨  

👍

File: packages/calcite-components/src/components/table-cell/table-cell.scss (hunk 1 above)
Comment (macandcheese) [reply packages/calcite-components/src/components/table-cell/table-cell.scss:14]: Agreed, added a class for this.

**Pull request merged**